src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -0,0 +1,216 @@
+using Standard;
+using System.Windows.Appearance;
+using System.Windows.Media;
+using Microsoft.Win32;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows.Interop;
+
+namespace System.Windows;
+
+internal static class ThemeManager
+{
+
+    #region Constructor
+
+    static ThemeManager()
+    {
+        _cachedThemeDictionaryUris = new List<Uri>();
+
+        if (Application.Current != null)
+        {
+            foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
+            {
+                if (mergedDictionary.Source != null && mergedDictionary.Source.ToString().EndsWith("FluentWindows.xaml"))
+                {
+                    _isFluentWindowsThemeEnabled = true;
+                    break;
+                }
+            }
+        }
+
+        if(_isFluentWindowsThemeEnabled)
+        {
+            _currentApplicationTheme = GetSystemTheme();
+            DwmColorization.UpdateAccentColors();
+        }
+    }
+
+    #endregion
+
+    #region Internal Methods
+
+    internal static void ApplySystemTheme(bool forceUpdate = false)
+    {
+        string systemTheme = GetSystemTheme();
+        bool useLightMode = GetUseLightTheme();
+        Color systemAccentColor = DwmColorization.GetSystemAccentColor();
+
+        var windows = Application.Current?.Windows;
+        if(windows != null)
+        {
+            ApplyTheme(windows , systemTheme, useLightMode, systemAccentColor, forceUpdate);
+        }
+    }
+
+    internal static void ApplySystemTheme(Window window, bool forceUpdate = false)
+    {
+        string systemTheme = GetSystemTheme();
+        bool useLightMode = GetUseLightTheme();
+        Color systemAccentColor = DwmColorization.GetSystemAccentColor();
+        ApplyTheme(new ArrayList { window } , systemTheme, useLightMode, systemAccentColor, forceUpdate);
+    }
+
+    internal static void ApplyTheme(
+        IEnumerable windows, 
+        string requestedTheme, 
+        bool requestedUseLightMode,
+        Color requestedAccentColor, 
+        bool forceUpdate = false)
+    {
+        bool needsUpdate = forceUpdate;
+        
+        if(requestedUseLightMode != _currentUseLightMode || DwmColorization.GetSystemAccentColor() != DwmColorization.CurrentApplicationAccentColor)
```

Comment:
No point checking if `forceUpdate` is true. Also the other way round, don't you need to call `UpdateAccentColors()` when `forceUpdate` is `true`? If not, why do you even need to call it, are you not going to invalidate it by applying the theme?

Reply:
It's the other way around, missed adding the forceUpdate check for accent color. Fixing it.